enhancements/network/ovs-dynamic-cpu-pinning.md

```diff
@@ -0,0 +1,116 @@
+---
+title: ovs-dynamic-cpu-pinning
+authors:
+  - "@zeeke"
+reviewers:
+  - "@trozet"
+  - "@MarSik"
+approvers:
+  - TBD
+creation-date: 2023-03-31
+last-updated: 2023-03-31
+tracking-link:
+  - https://issues.redhat.com/browse/CNF-5977
+---
+
+# OVS Dynamic CPU Pinning
+
+## Summary
+
+Allow OVS daemons to run on non reserved CPU if the network workload requires
+cycles and non-reserved CPU are idle.
+
+## Motivation
+
+OVS runs at reserved cpus when an OCP node is configured cpuManagerPolicy to `static`. (TBD doclink)
+Typically the customers want to minimize the amount of reserved cpus and leave
+as much cpus as possible available for the business related workloads.
+
+We have seen in some customer cases that it is possible to disrupt both the node
+and the cluster when the networking load rises, because OVS cannot keep up with 
+the flow and starts removing routes from its tables. Which just makes the situation 
+worse as the amount of upcalls (software based processing when a connection is not 
+present in the OVS tables) increases and increases the cpu load even further.
+
+
+### User Stories
+
+> As an Openshift cluster administrator
+> I want to make the most of the node's CPUs
+> So that the cluster can endure a high network workload.
+
+### Goals
+
+Every cluster node that is configured with static CPU manager policy and
+a low number of reserved CPU does not experience any excessive latency 
+increase while dealing with high network loads.
+
+### Non-Goals
+
+TBD
+
+## Proposal
+
+The main idea to resolve the issue described above is to allow OVS 
+to spread out to other cpus outside the reserved cpu pool. However 
+a care must be taken to not interfere with cpus that are assigned 
+to latency sensitive workloads.
```

Comment:
Is it NTO that should know where latency-sensitive workloads are running? Are we addressing this already?
Since we're not describing the details of NTO, maybe this is a non-goal for this enhancement actually.

Reply:
I guess latency-sensitive workload is the set of Pod that run in the Guaranteed Quality of Service. @MarSik if you confirm that I'll add a note to the doc.

Reply:
kubelet's cpu manager manages the cpu assignments and makes sure that the burstable pods will not interfere. This enhancement describes how OVN-K will transfer the mask from burstable containers (managed by cpu manager) to OVS (managed by OVN-K).